module-check/src/main/clojure/clojure/core/typed/base_env.clj

```diff
@@ -1443,7 +1443,14 @@ clojure.core/rand-nth (All [x] [(U (Indexed x) (SequentialSeqable x)) -> x])
   this-ns
 clojure.set/union (All [x] [(Set x) * -> (Set x)])
 clojure.set/intersection (All [x] [(Set x) (Set x) * -> (Set x)])
+
 clojure.set/difference (All [x] [(Set x) (Set Any) * -> (Set x)])
+clojure.set/subset?[(Set Any) (Set Any)  -> Boolean]
+clojure.set/superset? [(Set Any) (Set Any)  -> Boolean]
+clojure.set/join [(Set Any) (Set Any)  -> (Set Any)]
+clojure.set/index [(Set Any) (Vec Any)  -> (Map Any Any)]
```

Comment:
index should be `[(Set (Map Any Any)) (Vec Any) -> (Map (Map Any Any) (Set (Map Any Any)))]`.